src/controllers/profiles.js

```
@@ -1,57 +1,71 @@
-const fs = require("fs");
-const profiles = JSON.parse(fs.readFileSync("./data/profiles.json"));
+const Profiles = require("../models/profiles");
+const ObjectId = require("mongoose").Types.ObjectId;
 
-// Get Profiles
 const getAll = (req, res) => {
-  return res.status(200).json(profiles);
+  Profiles.find()
+    .then((profiles) => {
+      return res.status(200).json(profiles);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
+    });
+};
+
+const getById = (req, res) => {
+  Profiles.findById({ _id: new ObjectId(req.params.id) })
+    .then((profile) => {
+      return res.status(200).json(profile);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
+    });
 };
 
-// Create Profile
 const create = (req, res) => {
-  if (!req.query.id || !req.query.name || !req.query.description) {
-    return res.status(400).send("Some parameters are missing");
-  }
   const newProfile = {
-    id: (profiles.length + 1).toString(),
-    name: req.query.name,
-    description: req.query.description,
+    name: req.body.name,
+    description: req.body.description,
   };
+  Profiles.create(newProfile);
   return res.status(201).json(newProfile);
```

@ValentinaCollareta we should also return an error response. We could use the same promise that was used in getAll & getById function.